spring-cloud-stream-metrics/src/main/java/org/springframework/cloud/stream/metrics/config/MetricJsonSerializer.java

```diff
@@ -70,16 +73,28 @@ public void serialize(Metric<?> metric, JsonGenerator json,
 			String name = node.get("name").asText();
 			Number value = node.get("value").asDouble();
 			Date timestamp = null;
+			DateFormat df = formattersQueue.poll();
 
+			if(df == null){
+				df = defaultDateFormat();
+			}
 			try {
 				timestamp = df.parse(node.get("timestamp").asText());
 			}
 			catch (ParseException e) {
+			}finally {
+				formattersQueue.offer(df);
 			}
 			Metric<Number> metric = new Metric<Number>(name, value, timestamp);
 
 			return metric;
 		}
 
 	}
+
+	public static DateFormat defaultDateFormat(){
+		DateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSX");
+		df.setTimeZone(TimeZone.getTimeZone("GMT"));
+		return df;
```

Comment:
Actually i'll take that back, just remembered why we did it in the first place (DF is not thread safe, thus can not be shared while in the middle of formatting)